stdenv: Rename `isGNU` to `isHurd` as GNU is a userland

Ericson2314 opened the pull request:
###### Motivation for this change

Elsewhere, things called GNU indeed includes GNU/Linux or GNU/Hurd, but this predicate was defined excluding Linux regardless of userland.


###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

CC @matthewbauer (it won't let me request a review)

Ericson2314 committed:
stdenv: Rename `isGNU` to `isHurd` as GNU is a userland

Elsewhere, things called GNU indeed includes GNU/Linux or GNU/Hurd, but this
predicate was defined excluding Linux regardless of userland.

matthewbauer commented on pkgs/stdenv/generic/default.nix:
```diff
@@ -402,13 +402,10 @@ let
       isAarch64 = hostPlatform.isArm64;
-      # Other code instead checks for anything using GNU userland,
-      # e.g. GNU/linux. This refers just to GNU Hurd.
-      isGNU = system == "i686-gnu";
```
There might be a few references to this you missed:

- https://github.com/NixOS/nixpkgs/blob/9d5c28d22cf6573da2f75da2b9ec67f351c92f7d/pkgs/development/tools/misc/gdb/default.nix#L20-L22
- https://github.com/NixOS/nixpkgs/blob/9d5c28d22cf6573da2f75da2b9ec67f351c92f7d/pkgs/tools/archivers/zpaq/zpaqd.nix#L12

Ericson2314 replied:
Thanks!!

Ericson2314 committed:
gdb: Modernize and simplify derivation

 - `isGNU` to `isHurd`, my sedding did not catch
 - Simplify cross compilation in general
 - Be more careful about python
 - no more `gdbCross` in all-packages

Ericson2314 committed:
zpaqd: Modernize derivation

 - Simplified platform-specific options
 - Almost ready for cross-compilation

Ericson2314 commented:
Travis is borked from some big rebuild, so merging.
